File: 0053-maximum-subarray/0053-maximum-subarray.cpp
class Solution {
public:
    int maxSubArray(vector<int>& nums) {
        int maxi = INT_MIN;
        int curr = 0;
        for(auto num: nums) {
            curr = max(num, curr + num);
            maxi = max(maxi, curr);
        }
        return maxi;
    }
};

File: 0053-maximum-subarray/0053-maximum-subarray.js
/**
 * @param {number[]} nums
 * @return {number}
 */
var maxSubArray = function(nums) {
    let res = nums[0];
    let maxSoFar = nums[0];

    for(let i = 1; i < nums.length; i++) {
        maxSoFar = Math.max(maxSoFar + nums[i], nums[i]);
        res = Math.max(maxSoFar, res);
    }
    return res;
};